### PR TITLE
Validate configs.2

### PR DESCRIPTION
**Pull request by meejah**
A version of #505 with a shorter branch name (!!) to see if that makes windows less-sad about path lengths...

### CODE REVIEW

**Comment by exarkun** [comment]
I think AppVeyor didn't actually do a re-build for this PR.  It noticed the HEAD revision was the same as a previous build and just reported those results.

I'm going to push a junk commit to trigger a new build to see if that makes a difference (but I don't think it's going to, I'm pretty sure I've seen PRs for branches with much longer names pass on AppVeyor).

**Comment by exarkun** [comment]
Or maybe I'll use the AppVeyor "RE-BUILD PR" button and see what that does, first.

**Comment by exarkun** [comment]
The "RE-BUILD PR" build: https://ci.appveyor.com/project/tahoe-lafs/tahoe-lafs/build/1.0.1281 - looks like it has the same failures.

**Comment by exarkun** [comment]
The new master build: https://ci.appveyor.com/project/tahoe-lafs/tahoe-lafs/build/1.0.1282 - it does _not_ seem to have the same problems.

Hrm.

File: src/allmydata/test/cli/test_start.py [hunk 1]
```diff
@@ -207,3 +208,49 @@ def _exit(code):
...
+        # should print out the collected logs and an error-code
+        self.assertIn(
+            "invalid section",
+            o.getvalue(),
+        )
+        # this is SystemExit(0) for some reason I can't understand,
+        # while running on the command-line, "echo $?" shows "1" on
+        # this same error (some config exception)...
+        errs = self.flushLoggedErrors(SystemExit)
+        self.assertEqual(1, len(errs))
```
[the remark this answers is not in the document]
Comment (meejah) [reply src/allmydata/test/cli/test_start.py]: The other weird part is it's SystemExit with code=0, but CLI tests do exit with 1 so .... :woman_shrugging:

File: src/allmydata/test/cli/test_start.py [hunk 2]
```diff
@@ -207,3 +209,69 @@ def _exit(code):
...
+            "Configuration error:",
+            output,
+        )
+        # this is SystemExit(0) for some reason I can't understand,
+        # while running on the command-line, "echo $?" shows "1" on
+        # this same error (some config exception)...
+        errs = self.flushLoggedErrors(SystemExit)
+        self.assertEqual(1, len(errs))
+        # ensure reactor.stop was actually called
+        self.assertEqual([None], stopped)
```
Comment (meejah) [inline src/allmydata/test/cli/test_start.py]: Added some more asserts to confirm we are calling `reactor.stop`

File: src/allmydata/scripts/tahoe_daemonize.py [hunk 3]
```diff
@@ -125,8 +126,12 @@ def start():
...
                 raise ValueError("unknown nodetype %s" % self.nodetype)
 
-            srv = service_factory()
-            srv.setServiceParent(self.parent)
+            try:
+                srv = service_factory()
+                srv.setServiceParent(self.parent)
+            except UnknownConfigError as e:
+                sys.stderr.write("\nConfiguration error:\n{}\n\n".format(e))
+                reactor.stop()
```
[the remark this answers is not in the document]
Comment (meejah) [reply src/allmydata/scripts/tahoe_daemonize.py]: (see below, added test for this)

**Pull request merged**